### ISSUE TITLE
代码请求

### ISSUE THREAD

**Issue by akang-ai** [LiamLian0727/USIS10K#12]
## ---------------------- DEFAULT_SETTING ----------------------

default_scope = 'mmdet'
custom_imports = dict(imports=['project.our.our_model'], allow_failed_imports=False)

default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=20),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=3, max_keep_ckpts=1, save_best=['coco/bbox_mAP', 'coco/segm_mAP'],
                    rule='greater', save_last=True),
    sampler_seed=dict(type='DistSamplerSeedHook'),
)

env_cfg = dict(
    cudnn_benchmark=False,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl')
)

vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(type='DetLocalVisualizer', vis_backends=vis_backends, name='visualizer')
log_processor = dict(type='LogProcessor', window_size=50, by_epoch=True)

log_level = 'INFO'
load_from = None
resume = False

## ---------------------- MODEL ----------------------

crop_size = (1440, 810)

batch_augments = [
    dict(
        type='BatchFixedSizePad',
        size=crop_size,
        img_pad_value=0,
        pad_mask=True,
        mask_pad_value=0,
        pad_seg=False)
]

data_preprocessor = dict(
    type='DetDataPreprocessor',
    mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
    std=[0.229 * 255, 0.224 * 255, 0.225 * 255],
    bgr_to_rgb=True,
    pad_mask=True,
    pad_size_divisor=32,
    batch_augments=batch_augments
)

num_classes = 4
pointset_point_num = 5  # per pointset point

# sam base model
sam_pretrain_name = r"E:\Model\USIS10K-main\sam-vit-base"
sam_pretrain_ckpt_path = r"E:\Model\USIS10K-main\sam-vit-base\pytorch_model.bin"

# model settings
model = dict(
    type='USISAnchor',
    data_preprocessor=data_preprocessor,
    decoder_freeze=True,
    shared_image_embedding=dict(
        type='USISSamPositionalEmbedding',
        hf_pretrain_name=sam_pretrain_name,
        init_cfg=dict(type='Pretrained', checkpoint=sam_pretrain_ckpt_path),
    ),
    backbone=dict(
        type='USISSamVisionEncoder',
        hf_pretrain_name=sam_pretrain_name,
        extra_config=dict(output_hidden_states=True),
        init_cfg=dict(type='Pretrained', checkpoint=sam_pretrain_ckpt_path),
        peft_config=None,
    ),
    adapter=dict(
        type='UAViTAdapters', adapter_layer=range(5, 13, 2), embed_dim=768
    ),
    #### 'adapter_config' should be changed when using different pretrain model
    neck=dict(
        type='USISFPN',
        feature_aggregator=dict(
            type='USISFeatureAggregator',
            in_channels=[768] * (12 + 1),
            #### 'in_channels' should be changed when using different pretrain model
            # base:[768] * (12 + 1), large:[1024] * (24 + 1), huge:[1280] * (32 + 1)
            out_channels=128,
            hidden_channels=16,
            select_layers=range(5, 13, 2),
            #### 'select_layers' should be changed when using different pretrain model
            # base: range(1, 13, 2), large: range(1, 25, 2), huge: range(1, 33, 2)
        ),
        feature_spliter=dict(
            type='USISSimpleFPNHead',
            backbone_channel=128,
            in_channels=[32, 64, 128, 128],
            out_channels=128,
            num_outs=5,
            norm_cfg=dict(type='LN2d', requires_grad=True)),
    ),
    rpn_head=dict(
        type='RPNHead',
        in_channels=128,
        feat_channels=128,
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[4, 8 ],
            ratios=[0.5, 1.0, 2.0],
            strides=[4,8, 16, 32,64]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[.0, .0, .0, .0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        loss_bbox=dict(type='SmoothL1Loss', loss_weight=1.0)),
    roi_head=dict(
        type='USISPrompterAnchorRoIPromptHead',
        with_extra_pe=True,
        bbox_roi_extractor=dict(
            type='SingleRoIExtractor',
            roi_layer=dict(type='RoIAlign', output_size=7, sampling_ratio=0),
            out_channels=128,
            featmap_strides=[ 16, 32]),
        bbox_head=dict(
            type='Shared2FCBBoxHead',
            in_channels=128,
            fc_out_channels=512,
            roi_feat_size=7,
            num_classes=num_classes,
            bbox_coder=dict(
                type='DeltaXYWHBBoxCoder',
                target_means=[0., 0., 0., 0.],
                target_stds=[0.1, 0.1, 0.2, 0.2]),
            reg_class_agnostic=False,
            loss_cls=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
            loss_bbox=dict(type='SmoothL1Loss', loss_weight=1.0)),
        mask_roi_extractor=dict(
            type='SingleRoIExtractor',
            roi_layer=dict(type='RoIAlign', output_size=14, sampling_ratio=0),
            out_channels=128,
            featmap_strides=[4,8,16, 32]),
        mask_head=dict(
            type='USISPrompterAnchorMaskHead',
            mask_decoder=dict(
                type='USISSamMaskDecoder',
                hf_pretrain_name=sam_pretrain_name,
                init_cfg=dict(type='Pretrained', checkpoint=sam_pretrain_ckpt_path)),
            in_channels=128,
            roi_feat_size=7,
            per_pointset_point=pointset_point_num,
            with_sincos=True,
            multimask_output=False,
            class_agnostic=True,
            loss_mask=dict(type='CrossEntropyLoss', use_mask=True, loss_weight=1.0)
        )
    ),
    # model training and testing settings
    train_cfg=dict(
        rpn=dict(
            assigner=dict(
                type='MaxIoUAssigner',
                pos_iou_thr=0.7,
                neg_iou_thr=0.3,
                min_pos_iou=0.3,
                match_low_quality=True,
                ignore_iof_thr=-1),
            sampler=dict(
                type='RandomSampler',
                num=128,
                pos_fraction=0.5,
                neg_pos_ub=-1,
                add_gt_as_proposals=False),
            allowed_border=-1,
            pos_weight=-1,
            debug=False),
        rpn_proposal=dict(
            nms_pre=1000,
            max_per_img=500,
            nms=dict(type='nms', iou_threshold=0.7),
            min_bbox_size=0),
        rcnn=dict(
            assigner=dict(
                type='MaxIoUAssigner',
                pos_iou_thr=0.5,
                neg_iou_thr=0.5,
                min_pos_iou=0.5,
                match_low_quality=True,
                ignore_iof_thr=-1),
            sampler=dict(
                type='RandomSampler',
                num=128,
                pos_fraction=0.2,
                neg_pos_ub=-1,
                add_gt_as_proposals=True),
            mask_size=crop_size,
            pos_weight=-1,
            debug=False)),
    test_cfg=dict(
        rpn=dict(
            nms_pre=500,
            max_per_img=200,
            nms=dict(type='nms', iou_threshold=0.7),
            min_bbox_size=0),
        rcnn=dict(
            score_thr=0.05,
            nms=dict(type='nms', iou_threshold=0.5),
            max_per_img=100,
            mask_thr_binary=0.5)
    )
)
以上代码是我的网络结构

_base_ = ['anchor_net.py']

work_dir = './work_dirs'

## ---------------------- MODEL ----------------------

crop_size = (1440, 810)
num_classes = 4

batch_augments = [dict(
    type='BatchFixedSizePad',
    size=crop_size,
    img_pad_value=0,
    pad_mask=True,
    mask_pad_value=0,
    pad_seg=False
)]

data_preprocessor = dict(
    type='DetDataPreprocessor',
    mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
    std=[0.229 * 255, 0.224 * 255, 0.225 * 255],
    bgr_to_rgb=True,
    pad_mask=True,
    pad_size_divisor=32,
    batch_augments=batch_augments
)

model = dict(
    data_preprocessor=data_preprocessor,
    decoder_freeze=True,
    shared_image_embedding=dict(extra_config=dict(image_size=crop_size[0])),
    backbone=dict(extra_config=dict(image_size=crop_size[0])),
    roi_head=dict(bbox_head=dict(num_classes=num_classes)),
    train_cfg=dict(rcnn=dict(mask_size=crop_size))
)

## ---------------------- Dataset ----------------------

# dataset settings
dataset_type = 'MultiClassUSIS10KInsSegDataset'
data_root = 'data/liu'

backend_args = None

train_pipeline = [
    dict(type='LoadImageFromFile', backend_args=backend_args, to_float32=True),
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(type='RandomFlip', prob=0.5),
    # large scale jittering
    dict(
        type='RandomResize',
        scale=crop_size,
        ratio_range=(0.1, 2.0),
        resize_type='Resize',
        keep_ratio=True),
    dict(
        type='RandomCrop',
        crop_size=crop_size,
        crop_type='absolute',
        recompute_bbox=True,
        allow_negative_crop=True),
    dict(type='FilterAnnotations', min_gt_bbox_wh=(1e-5, 1e-5), by_mask=True),
    dict(type='PackDetInputs')
]

test_pipeline = [
    dict(type='LoadImageFromFile', backend_args=backend_args, to_float32=True),
    dict(type='Resize', scale=(2560,1440), keep_ratio=True),
    dict(type='Pad', size=(2560,1440), pad_val=dict(img=(0.406 * 255, 0.456 * 255, 0.485 * 255), masks=0)),
    # If you don't have a gt annotation, delete the pipeline
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(
        type='PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape', 'pad_shape', 'scale_factor')
    )
]

batch_size = 1
num_workers = 4
persistent_workers = True
indices = None

#CLASSES = ['wrecks/ruins', 'fish', 'reefs', 'aquatic plants', 'human divers', 'robots', 'sea-floor']
#CLASSES = dict(classes=['wrecks/ruins', 'fish', 'reefs', 'aquatic plants', 'human divers', 'robots', 'sea-floor'])
CLASSES = dict(classes=['barricade', 'brick', 'shrimp', 'fish'])

train_dataloader = dict(
    batch_size=batch_size,
    num_workers=num_workers,
    persistent_workers=persistent_workers,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        indices=indices,
        data_root=data_root,
        metainfo=CLASSES,
        ann_file='annotations/train.json',
        data_prefix=dict(img='train'),
        # filter_cfg=dict(filter_empty_gt=True, min_size=32),
        pipeline=train_pipeline,
        backend_args=backend_args)
)

val_dataloader = dict(
    batch_size=batch_size,
    num_workers=num_workers,
    persistent_workers=persistent_workers,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        indices=indices,
        data_root=data_root,
        metainfo=CLASSES,
        ann_file='annotations/val.json',
        data_prefix=dict(img='val'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args)
)

test_dataloader = dict(
    batch_size=batch_size,
    num_workers=num_workers,
    persistent_workers=persistent_workers,
    drop_last=False,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        indices=indices,
        data_root=data_root,
        metainfo=CLASSES,
        ann_file='annotations/test.json',
        data_prefix=dict(img='test'),
        test_mode=True,
        pipeline=test_pipeline,
        backend_args=backend_args)
)

val_evaluator = dict(
    type='CocoMetric',
    metric=['bbox', 'segm'],
    ann_file=data_root + '/annotations/val.json',
    format_only=False,
    backend_args=backend_args,
)

test_evaluator = dict(
    type='CocoMetric',
    metric=['bbox', 'segm'],
    ann_file=data_root + '/annotations/test.json',
    format_only=False,
    backend_args=backend_args,
)

## ---------------------- Optim ----------------------

max_epochs = 24
base_lr = 0.0001

train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=max_epochs, val_interval=3)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')

find_unused_parameters = True

# learning rate
param_scheduler = [
    dict(type='LinearLR', start_factor=0.001, by_epoch=False, begin=0, end=50),
    dict(
        type='MultiStepLR',
        begin=0,
        end=max_epochs,
        by_epoch=True,
        milestones=[15, 21],
        gamma=0.1
    )
]

optim_wrapper = dict(
    type='AmpOptimWrapper',
    dtype='float16',
    optimizer=dict(
        type='AdamW',
        lr=base_lr,
        weight_decay=0.05,
    )
)

# Default setting for scaling LR automatically
#   - `enable` means enable scaling LR automatically
#       or not by default.
#   - `base_batch_size` = (8 GPUs) x (2 samples per GPU).
auto_scale_lr = dict(enable=False, base_batch_size=batch_size)

#default_hooks = dict(visualization=dict(type='VisualizationHook', interval=1))# Customize interval as needed
以上代码是我的配置文件，我使用的是sam-vit-base作为我的模型（由于自己电脑显存不够，且自己制作的数据集比较简单），为什么老是报出找不到数据集里的图片这样的错误，例如这样
![image](https://github.com/user-attachments/assets/8c979830-b942-4aef-a647-825af1ddcb3d)

**Comment by LiamLian0727** [comment]
This error is caused by the fact that the path to the image corresponding to your training set does not exist.
First, you can check whether the path `data/liu\\train\\607.jpg` exists, and second, you can replace `\\` in the path with `/`.

**Comment by akang-ai** [comment]
![image](https://github.com/user-attachments/assets/153ef1d1-24bf-474e-9c76-4a357d014fde)
![image](https://github.com/user-attachments/assets/7e8b2672-a247-46c5-89e5-14c3f1b9bba2)
![image](https://github.com/user-attachments/assets/fb7467e2-19d0-49bc-a2f8-166b6644650c)
![image](https://github.com/user-attachments/assets/d81cf415-b698-4ec9-949d-c82e10c983bb)
![image](https://github.com/user-attachments/assets/60ce76c7-8f3f-4ddc-a1ac-2ea5264327e9)
都是存在的

**Comment by LiamLian0727** [comment]
Your dataroot is `data\liu` and data_prefix  is `train`. So the train dataset path in mmengine is `data\liu\train`.
However, observing in the image you gave, it seems that no such path exists

**Comment by LiamLian0727** [comment]
Additionally, I've noticed that you've been frequently starting different issues in my repository for the same problems you've been shown, and while I appreciate you recognizing the work we've been doing, I urge you to double-check that your configurations make sense, make sure you're not making mistakes by checking the documentation, and post questions that share the same topic under one issue instead of multiple. Thank you for understanding.
